searx/engines/google_videos.py

```diff
@@ -114,40 +98,59 @@ def response(resp):
-        content = extract_text(c_node)
-        pub_info = extract_text(eval_xpath(result, './/div[contains(@class, "gqF9jc")]'))
+    result_divs = eval_xpath_list(dom, '//div[contains(@class, "MjjYud")]')
 
-        results.append(
-            {
+    # parse results
+    for i, result in enumerate(result_divs):
+        title = extract_text(eval_xpath_getindex(result, './/h3[contains(@class, "LC20lb")]', 0, default=None), allow_none=True)
+        url = eval_xpath_getindex(result, './/a[@jsname="UWckNb"]/@href', 0, default=None)
+        content = extract_text(eval_xpath_getindex(result, './/div[contains(@class, "ITZIwc")]', 0, default=None), allow_none=True)
+        pub_info = extract_text(eval_xpath_getindex(result, './/div[contains(@class, "gqF9jc")]', 0, default=None), allow_none=True)
+        # Broader XPath to find any <img> element
+        thumbnail = eval_xpath_getindex(result, './/img/@src', 0, default=None)
+        duration = extract_text(eval_xpath_getindex(result, './/span[contains(@class, "k1U36b")]', 0, default=None), allow_none=True)
+        video_id = eval_xpath_getindex(result, './/div[@jscontroller="rTuANe"]/@data-vid', 0, default=None)
+
+        # Handle thumbnail
+        if thumbnail and thumbnail.startswith('data:image'):
+            img_id = eval_xpath_getindex(result, './/img/@id', 0, default=None)
+            if img_id and img_id in data_image_map:
+                thumbnail = data_image_map[img_id]
+            else:
+                thumbnail = None
+        # Attempt YouTube fallback even if thumbnail is None
+        if not thumbnail and (video_id or (url and 'youtube.com' in url)):
+            if video_id:
+                thumbnail = f"https://img.youtube.com/vi/{video_id}/hqdefault.jpg"
+            elif url:
+                from urllib.parse import urlparse, parse_qs
+                parsed_url = urlparse(url)
+                video_id_from_url = parse_qs(parsed_url.query).get('v', [None])[0]
+                if video_id_from_url:
+                    thumbnail = f"https://img.youtube.com/vi/{video_id_from_url}/hqdefault.jpg"
+
+        # Handle video embed URL
+        embed_url = None
+        if video_id:
+            embed_url = get_embeded_stream_url(f"https://www.youtube.com/watch?v={video_id}")
+        elif url:
+            embed_url = get_embeded_stream_url(url)
+
+        # Only append results with valid title and url
+        if title and url:
+            results.append({
                 'url': url,
                 'title': title,
-                'content': content,
-                'author': pub_info,
-                'thumbnail': thumbnail,
-                'iframe_src': get_embeded_stream_url(url),
+                'content': content or '',
+                'author': pub_info or '',
+                'thumbnail': thumbnail or '',
+                'length': duration or '',
+                'iframe_src': embed_url or '',
```

Comment:
```suggestion
                'content': content or '',
                'author': pub_info,
                'thumbnail': thumbnail,
                'length': duration,
                'iframe_src': embed_url,
```
It's fine if they're `None`, only `content` may not be `None`.

Reply:
Thank you for the feedback @Bnyro , I implemented all your suggestions in the last commit. Rest of the code remains unchanged.